pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -595,11 +594,13 @@ func (c *Cluster) getOSDInfo(d *appsv1.Deployment) (OSDInfo, error) {
 	}
 
 	locationFound := false
-	for _, a := range container.Command {
+	for _, a := range container.Args {
```

Comment:
How about moving this `for` loop inside a helper method and then keep a unit test for it? String parsing is always a place I'd recommend unit tests.